pkg/pulumiyaml/analyser.go

```diff
@@ -455,7 +455,7 @@ func (tc *typeCache) typeResource(r *runner, node resourceNode) bool {
 	}
 	if s := v.Syntax(); s != nil {
 		if o, ok := s.(*syntax.ObjectNode); ok {
-			validKeys := append(v.Fields(), "condition", "metadata")
```

Comment:
I think this can be removed here, as `v.Fields()` should contain `defaultprovider`?